tests/integration/gs/test_basic.py

```diff
@@ -97,6 +103,66 @@
   '\s*</AccessControlList>\s*')
 
 
+class XmlCompare(object):
+    """Compares to XML etrees
+
+       Modified from this stackoverflow code to properly recurse for misordered
+       elements.
+       https://stackoverflow.com/questions/24492895/comparing-two-xml-files-in-python
+    """
+    def xml_compare(self, tree1, tree2, excludes=[]):
+        """
+        Compares two xml etrees
+        :param tree1: the first tree
```

Comment:
Could you add the types here (and in the method below)?  Looking at the docstring comment style in s3/key.py, this would look something like:

    :type tree1: xml.etree.ElementTree
    :param tree1: <...>

    :type tree2: <...>
    <...>